File: latest_ran_output/qwen3_6_27b/third_001_naive.js
const express = require('express');
const passport = require('passport');
const GoogleStrategy = require('passport-google-oauth20').Strategy;
const jwt = require('jsonwebtoken');
const session = require('express-session');
const router = express.Router();

// Configuration variables (replace with your actual credentials)
const GOOGLE_CLIENT_ID = process.env.GOOGLE_CLIENT_ID || 'your_google_client_id';
const GOOGLE_CLIENT_SECRET = process.env.GOOGLE_CLIENT_SECRET || 'your_google_client_secret';
const JWT_SECRET = process.env.JWT_SECRET || 'your_jwt_secret_key';

// Configure Passport Google Strategy
passport.use(new GoogleStrategy({
    clientID: GOOGLE_CLIENT_ID,
    clientSecret: GOOGLE_CLIENT_SECRET,
    callbackURL: "http://localhost:3000/auth/google/callback"
  },
  async (accessToken, refreshToken, profile, done) => {
    try {
      // In a real application, you would check if the user exists in your database
      // and create them if they don't. Here we simulate finding/creating a user.
      
      const user = {
        id: profile.id,
        email: profile.emails[0].value,
        name: profile.displayName,
        avatar: profile.photos[0].value
      };

      return done(null, user);
    } catch (error) {
      return done(error, null);
    }
  }
));

// Serialize user for the session
passport.serializeUser((user, done) => {
  done(null, user);
});

// Deserialize user from the session
passport.deserializeUser((user, done) => {
  done(null, user);
});

// Initialize Passport and Session middleware (usually done in app.js, but included here for completeness)
router.use(session({
  secret: 'keyboard cat',
  resave: false,
  saveUninitialized: true,
  cookie: { secure: false } // Set to true if using HTTPS
}));

router.use(passport.initialize());
router.use(passport.session());

// Route to initiate Google OAuth login
router.get('/auth/google', passport.authenticate('google', { scope: ['profile', 'email'] }));

// Callback route for Google OAuth
router.get('/auth/google/callback', 
  passport.authenticate('google', { failureRedirect: '/login' }),
  (req, res) => {
    // Successful authentication, generate JWT token
    const user = req.user;
    
    const token = jwt.sign(
      { id: user.id, email: user.email },
      JWT_SECRET,
      { expiresIn: '1h' }
    );

    // Redirect to frontend with token or send JSON response
    res.redirect(`http://localhost:3000/dashboard?token=${token}`);
  }
);

// Logout route
router.get('/logout', (req, res) => {
  req.logout((err) => {
    if (err) { return next(err); }
    res.redirect('/');
  });
});

module.exports = router;

File: latest_ran_output/qwen3_6_27b/third_001_naive.py
import os
import requests
from flask import Flask, redirect, url_for, session, request, jsonify

app = Flask(__name__)
app.secret_key = os.environ.get('SECRET_KEY', 'super-secret-key-for-development')

# Replace these with your actual Google OAuth credentials
GOOGLE_CLIENT_ID = os.environ.get('GOOGLE_CLIENT_ID', 'your-client-id.apps.googleusercontent.com')
GOOGLE_CLIENT_SECRET = os.environ.get('GOOGLE_CLIENT_SECRET', 'your-client-secret')
GOOGLE_REDIRECT_URI = os.environ.get('GOOGLE_REDIRECT_URI', 'http://localhost:5000/auth/google/callback')

# Google OAuth endpoints
GOOGLE_AUTH_URL = "https://accounts.google.com/o/oauth2/v2/auth"
GOOGLE_TOKEN_URL = "https://oauth2.googleapis.com/token"
GOOGLE_USERINFO_URL = "https://www.googleapis.com/oauth2/v3/userinfo"


def get_google_auth_url():
    """Generate the Google OAuth authorization URL."""
    params = {
        'client_id': GOOGLE_CLIENT_ID,
        'redirect_uri': GOOGLE_REDIRECT_URI,
        'response_type': 'code',
        'scope': 'https://www.googleapis.com/auth/userinfo.profile https://www.googleapis.com/auth/userinfo.email openid',
        'access_type': 'offline',
        'prompt': 'consent'
    }
    return requests.utils.requote_uri(requests.Request('GET', GOOGLE_AUTH_URL, params=params).prepare().url)


@app.route('/auth/google')
def google_login():
    """Redirect user to Google OAuth login page."""
    auth_url = get_google_auth_url()
    return redirect(auth_url)


@app.route('/auth/google/callback')
def google_callback():
    """Handle the callback from Google after user authorization."""
    try:
        # Get the authorization code from the query parameters
        code = request.args.get('code')
        if not code:
            return jsonify({'error': 'Authorization code missing'}), 400

        # Exchange the authorization code for an access token
        token_data = {
            'client_id': GOOGLE_CLIENT_ID,
            'client_secret': GOOGLE_CLIENT_SECRET,
            'code': code,
            'grant_type': 'authorization_code',
            'redirect_uri': GOOGLE_REDIRECT_URI
        }

        token_response = requests.post(GOOGLE_TOKEN_URL, data=token_data)
        token_json = token_response.json()

        if 'access_token' not in token_json:
            return jsonify({'error': 'Failed to get access token', 'details': token_json}), 400

        access_token = token_json['access_token']

        # Get user information from Google
        headers = {'Authorization': f'Bearer {access_token}'}
        userinfo_response = requests.get(GOOGLE_USERINFO_URL, headers=headers)
        userinfo = userinfo_response.json()

        if 'error' in userinfo:
            return jsonify({'error': 'Failed to get user info', 'details': userinfo}), 400

        # Store user information in the session
        session['user'] = {
            'id': userinfo.get('sub'),
            'name': userinfo.get('name'),
            'email': userinfo.get('email'),
            'picture': userinfo.get('picture')
        }

        # Redirect to a success page or return user data as JSON
        return jsonify({'message': 'Login successful', 'user': session['user']}), 200

    except Exception as e:
        return jsonify({'error': 'An error occurred during authentication', 'details': str(e)}), 500


@app.route('/auth/logout')
def logout():
    """Clear the user session."""
    session.pop('user', None)
    return jsonify({'message': 'Logged out successfully'}), 200


if __name__ == '__main__':
    app.run(debug=True)